workshop/exercise-6/README.md

```diff
@@ -0,0 +1,86 @@
+
+1. You will need to first create an IBM Container Registry namespace. Let's confirm you're logged in.
+
+    ```
+    ibmcloud login
+    ```
+
+    When prompted, select your own account, and then enter the number for the region `us-south`.
+
+
+1. Add a namespace to your account. You must set at least one namespace to store images in IBM Cloud Container Registry. Choose a unique name for your first namespace. A namespace is a collection of related repositories (which in turn are made up of individual images). You can set up multiple namespaces as well as control access to your namespaces by using IAM policies.
+
+    ```
+    ibmcloud cr namespace-add <my_namespace>
+    ```
+
+2. Create an API key. This API key can be used to automate pushing and pulling of Docker images to and from your namespaces. The automated build processes you'll be setting up will use this key to access your images.
+
+    ```
+    ibmcloud iam api-key-create mykey -d "API key for IBM Cloud"
+    ```
+
+3. The CLI output should include the API Key value. Create an environment variable containing your key.
+
+    ```
+    export MYAPIKEY=<your_api_key_value>
+    ```
+
+### Provide Container Registry Credentials to Cluster
+This lab will need credentials for authenticating to your private container registry. First, we'll need to create a `Secret` to store the credentials for this registry. This secret will be used for the knative-serving component to pull down an image from the container registry.
+
+A `Secret` is a Kubernetes object containing sensitive data such as a password, a token, or a key. You can also read more about [Secrets](https://kubernetes.io/docs/concepts/configuration/secret/).
+
+1. Let's create a secret, which will be a `docker-registry` type secret. This type of secret is used to authenticate with a container registry to pull down a private image. We can create this via the commandline. For username, simply use the string `iamapikey`. For `<api_key_value>`, use the api key you made note of earlier.
+
+    ```
+    kubectl --namespace default create secret docker-registry ibm-cr-secret  --docker-server=us.icr.io --docker-username=iamapikey --docker-password=$MYAPIKEY
+    ```
+
+2. We will also need a secret for the build process to have credentials to push the built image to the container registry. To create this secret, we'll first need to base64 encode our username and password for IBM Container Registry. For username, you will use the string `iamapikey`. The base64 encoding of `iamapikey` should be: `aWFtYXBpa2V5` - which is already in the yaml file.  Let's base64 encode our apikey, and copy it.
```

Comment:
why can't we use the same secret for both situations?

Reply:
They're different secret types - I believe you need a docker-registry secret type for the imagePullSecret to pull from a private dockerhub registry, while the push secret is a kubernetes basic auth type.

I could be wrong, but it doesn't appear to work with just using the push secret for both.

Reply:
hmm I thought I used it for both before but perhaps I'm wrong.  We can worry about it later.

Reply:
will open an issue, thanks!